app/controllers/annotation_categories_controller.rb

```diff
@@ -285,7 +285,22 @@ def annotation_text_uses
   end
 
   def uncategorized_annotations
+    @assignment = Assignment.find(params[:assignment_id])
+    @texts = annotation_text_data(nil)
+  end
+
+  def onetime_annotation_download
```

Comment:
Rather than create a separate route, you can use the same route as above (`uncategorized_annotations`) but have it respond to a different format (in this case, `csv`). You can do this using `respond_to` blocks. You can search for examples in our own code, and read the documentation [here](https://apidock.com/rails/ActionController/MimeResponds/respond_to).

It's a new idea, but basically you'll want a `respond_to.js` for the existing route, and a `respond_to.csv` with a "do block" for the new route.